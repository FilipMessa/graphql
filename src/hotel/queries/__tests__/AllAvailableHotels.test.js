// @flow

import {
  graphql,
  BookingComApiMock,
} from '../../../common/services/TestingTools';
import AllHotelsDataset from '../../datasets/all.json';
import FirstHotelDataset from '../../datasets/25215.json';
import SecondHotelDataset from '../../datasets/248539.json';
import RoomBlocksDataset from '../../datasets/roomBlocks.json';
import RoomDetailsDataset from '../../datasets/roomDetails.json';

beforeEach(() => {
  BookingComApiMock.onGet(
    'https://distribution-xml.booking.com/2.0/json/hotelAvailability?extras=hotel_details&order_by=popularity&radius=50&latitude=45.4654&longitude=9.1859&checkin=2017-11-16&checkout=2017-11-23&rows=50&room1=A%2CA%2C4%2C6&room2=A%2C2',
  ).replyWithData(AllHotelsDataset);
  BookingComApiMock.onGet(
    'https://distribution-xml.booking.com/2.0/json/hotels?extras=hotel_info%2Chotel_photos%2Chotel_description%2Chotel_facilities%2Cpayment_details%2Croom_info%2Croom_photos%2Croom_description%2Croom_facilities&hotel_ids=25215%2C248539&language=en-us',
  ).replyWithData({
    result: FirstHotelDataset.result.concat(SecondHotelDataset.result),
  });
  BookingComApiMock.onGet(
    'https://distribution-xml.booking.com/2.0/json/blockAvailability?detail_level=1&checkin=2017-11-16&checkout=2017-11-23&hotel_ids=25215&currency=EUR',
  ).replyWithData(RoomBlocksDataset);
  BookingComApiMock.onGet(
    // this can be merged into previous request (hotel_ids=25215,248539) but we are not doing this optimization yet
    'https://distribution-xml.booking.com/2.0/json/blockAvailability?detail_level=1&checkin=2017-11-16&checkout=2017-11-23&hotel_ids=248539&currency=EUR',
  ).replyWithData(RoomBlocksDataset);
  BookingComApiMock.onGet(
    'https://distribution-xml.booking.com/2.0/json/hotels?extras=hotel_info%2Chotel_photos%2Chotel_description%2Chotel_facilities%2Cpayment_details%2Croom_info%2Croom_photos%2Croom_description%2Croom_facilities&hotel_ids=25215&language=en-us',
  ).replyWithData(RoomDetailsDataset);
  BookingComApiMock.onGet(
    'https://distribution-xml.booking.com/2.0/json/hotels?extras=hotel_info%2Chotel_photos%2Chotel_description%2Chotel_facilities%2Cpayment_details%2Croom_info%2Croom_photos%2Croom_description%2Croom_facilities&hotel_ids=248539&language=en-us',
  ).replyWithData([]); // Booking.com failure
});

describe('all hotels query', () => {
  it('should work for full query', async () => {
    expect(
      await graphql(`
        {
          allAvailableHotels(
            search: {
              latitude: 45.4654
              longitude: 9.1859
              checkin: "2017-11-16"
              checkout: "2017-11-23"
              roomsConfiguration: [
                { adultsCount: 2, children: [{ age: 4 }, { age: 6 }] }
                { adultsCount: 1, children: [{ age: 2 }] }
              ]
            }
          ) {
            pageInfo {
              hasNextPage
              hasPreviousPage
              startCursor
              endCursor
            }
            edges {
              node {
                id
                price {
                  amount
                  currency
                }
                availableRooms {
                  id
                  originalId
                  minimalPrice {
                    amount
                    currency
                  }
                  incrementalPrice {
                    amount
                    currency
                  }
                  room {
                    id
                    type
                    roomSize
                  }
                }
                hotel {
                  id
                  originalId
                  summary
                  review {
                    score
                    description
                    count
                  }
                }
              }
            }
          }
        }
      `),
    ).toMatchSnapshot();
  });
});
