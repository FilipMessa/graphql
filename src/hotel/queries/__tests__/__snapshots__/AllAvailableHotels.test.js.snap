// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all hotels query should work for full query 1`] = `
Object {
  "data": Object {
    "allAvailableHotels": Object {
      "edges": Array [
        Object {
          "node": Object {
            "availableRooms": Array [
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI0ODUzOTAxXzk0MzY0OTI4XzFfMl8w",
                "incrementalPrice": Array [
                  Object {
                    "amount": 125,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 250,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 375,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 500,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 625,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 750,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 875,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 1000,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 125,
                  "currency": "EUR",
                },
                "originalId": "24853901_94364928_1_2_0",
                "room": null,
              },
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI0ODUzOTAxXzEwNjY2NjM4N18xXzJfMA==",
                "incrementalPrice": Array [
                  Object {
                    "amount": 125,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 250,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 375,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 500,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 625,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 750,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 875,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 1000,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 125,
                  "currency": "EUR",
                },
                "originalId": "24853901_106666387_1_2_0",
                "room": null,
              },
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI1MjE1MTVfOTUxNDkwMzZfMF8xXzA=",
                "incrementalPrice": Array [
                  Object {
                    "amount": 53.1,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 106.2,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 159.3,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 212.4,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 265.5,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 318.6,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 53.1,
                  "currency": "EUR",
                },
                "originalId": "2521515_95149036_0_1_0",
                "room": Object {
                  "id": "aG90ZWxSb29tOjI1MjE1MTU=",
                  "type": "Twin/Double",
                },
              },
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI1MjE1MTVfMTAzMDE5NjI5XzBfMV8w",
                "incrementalPrice": Array [
                  Object {
                    "amount": 59,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 118,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 177,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 236,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 295,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 354,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 59,
                  "currency": "EUR",
                },
                "originalId": "2521515_103019629_0_1_0",
                "room": Object {
                  "id": "aG90ZWxSb29tOjI1MjE1MTU=",
                  "type": "Twin/Double",
                },
              },
            ],
            "hotel": Object {
              "id": "aG90ZWw6MjUyMTU=",
              "originalId": "25215",
              "review": Object {
                "count": 2405,
                "description": null,
                "score": 8.3,
              },
              "summary": "Italiana Hotels Milan Rho Fair is set in a renovated cotton mill from the early 20th century, 10 minutes' drive from RhoFiera Milano exhibition centre. It offers air-conditioned rooms, free WiFi throughout, and a small gym with sauna and Turkish bath. This Italiana hotel features elegant rooms with parquet floors and a flat-screen TV with satellite channels. Some rooms feature tea and coffee making facilities, and luxury toiletries. Some also have a private terrace. On the top floor you will find the hotel's trendy Club 1902 bar, and the La Fabbrica del Cotone restaurant, both with panoramic views. The restaurant serves Mediterranean and Milanese specialities prepared by the chef each day. During major events, a free shuttle is offered to/from RhoFiera Milano exhibition centre. The property is 500 metres from Rho Train Station. From there, you can reach the EXPO 2015 exhibition centre in 4 minutes.",
            },
            "id": "SG90ZWxBdmFpbGFiaWxpdHk6MjUyMTU=",
            "price": Object {
              "amount": 528.71,
              "currency": "EUR",
            },
          },
        },
        Object {
          "node": Object {
            "availableRooms": Array [
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI0ODUzOTAxXzk0MzY0OTI4XzFfMl8w",
                "incrementalPrice": Array [
                  Object {
                    "amount": 125,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 250,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 375,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 500,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 625,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 750,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 875,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 1000,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 125,
                  "currency": "EUR",
                },
                "originalId": "24853901_94364928_1_2_0",
                "room": null,
              },
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI0ODUzOTAxXzEwNjY2NjM4N18xXzJfMA==",
                "incrementalPrice": Array [
                  Object {
                    "amount": 125,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 250,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 375,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 500,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 625,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 750,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 875,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 1000,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 125,
                  "currency": "EUR",
                },
                "originalId": "24853901_106666387_1_2_0",
                "room": null,
              },
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI1MjE1MTVfOTUxNDkwMzZfMF8xXzA=",
                "incrementalPrice": Array [
                  Object {
                    "amount": 53.1,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 106.2,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 159.3,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 212.4,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 265.5,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 318.6,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 53.1,
                  "currency": "EUR",
                },
                "originalId": "2521515_95149036_0_1_0",
                "room": Object {
                  "id": "aG90ZWxSb29tOjI1MjE1MTU=",
                  "type": "Twin/Double",
                },
              },
              Object {
                "id": "SG90ZWxSb29tQXZhaWxhYmlsaXR5OjI1MjE1MTVfMTAzMDE5NjI5XzBfMV8w",
                "incrementalPrice": Array [
                  Object {
                    "amount": 59,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 118,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 177,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 236,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 295,
                    "currency": "EUR",
                  },
                  Object {
                    "amount": 354,
                    "currency": "EUR",
                  },
                ],
                "minimalPrice": Object {
                  "amount": 59,
                  "currency": "EUR",
                },
                "originalId": "2521515_103019629_0_1_0",
                "room": Object {
                  "id": "aG90ZWxSb29tOjI1MjE1MTU=",
                  "type": "Twin/Double",
                },
              },
            ],
            "hotel": Object {
              "id": "aG90ZWw6MjQ4NTM5",
              "originalId": "248539",
              "review": Object {
                "count": 4781,
                "description": null,
                "score": 8.4,
              },
              "summary": "Sheraton Milan Malpensa Hotel is located inside Malpensa Airport's Terminal 1. It offers air-conditioned rooms with minibar and flat-screen TV. A fitness centre is also available. Rooms at Sheraton Milano Malpensa Airport Hotel and Conference Centre are soundproofed and feature Sheraton Sweet Sleeper beds. Other hotel facilities include business meeting rooms, a spa and beauty centre and indoor swimming pool. Guests can sip a cocktail at the on-site bar and have dinner at the restaurant, which specialises in Italian cuisine. The hotel is a 40-minute drive from Milan centre, and 40 km from Rho Fiera Milano Exhibition Centre. You can also use public and private shuttles to Milan Central Railway Station.",
            },
            "id": "SG90ZWxBdmFpbGFiaWxpdHk6MjQ4NTM5",
            "price": Object {
              "amount": 775,
              "currency": "EUR",
            },
          },
        },
      ],
    },
  },
}
`;
